src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Attributes.jl

```diff
@@ -583,3 +578,44 @@ function relative_cotangent_module(X::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing
   return cokernel(psi)
 end
 
+@doc raw"""
+    genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+
+Given a projective curve `X` return the genus of `X`, i.e. the 
+integer `p_g = p_a - delta` where `p_a` is the arithmetic genus 
+and `delta` the delta-invariant of the curve.
+
+The `algorithm` keyword can be specified to 
+  - `:normalization` to go via the computation of a normalization
+  - `:primary_decomposition` to proceed with a primary decomposition
+Normalization is usually faster, but not always.
+"""
+function genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+  @req dim(X) == 1 "scheme must be one-dimensional"
+  get_attribute!(X, :genus) do
+    I = defining_ideal(X)
+    I_sing = singular_generators(I)
+    if algorithm == :normalization
+      return Singular.LibNormal.genus(I_sing)
+    elseif algorithm == :primary_decomposition
+      return Singular.LibNormal.genus(I_sing, "prim")
+    else 
+      error("algorithm not recognized")
+    end
+  end::Int
+end
+
+@doc raw"""
+    arithmetic_genus(X::AbsProjectiveScheme{<:Field})
+
+Given an equidimensional projective curve `X` return the arithmetic genus of `X`, i.e. the 
```

Comment:
You could allow any variety, instead of a curve?

Reply:
shouldn't a curve be equidimensional by the definition?

Reply:
For the arithmetic genus, you need equidimensional, but it is classically defined for any dimension. So I would not mention 'curve' here.